generator/internal/rust/annotate.go

```diff
@@ -428,6 +431,15 @@ func (c *codec) addFeatureAnnotations(model *api.API, ann *modelAnnotations) {
 	}
 }
 
+// Maps "google.foo.v1" to "google::foo::v1"
+func packageToModuleName(p string) string {
+	components := strings.Split(p, ".")
+	for i, c := range components {
+		components[i] = toSnake(c)
+	}
+	return strings.Join(components, "::")
+}
+
```

Comment:
I am almost certain there is a function doing something similar in `codec.go`. Consider moving this function there.